treatment-service/config/config.go

```diff
@@ -24,16 +26,17 @@ type Config struct {
 	Port       int      `json:"port" default:"8080" validate:"required"`
 	ProjectIds []string `json:"project_ids" default:""`
 
-	AssignedTreatmentLogger AssignedTreatmentLoggerConfig       `json:"assigned_treatment_logger"`
-	DebugConfig             DebugConfig                         `json:"debug_config" validate:"required,dive"`
-	NewRelicConfig          newrelic.Config                     `json:"new_relic_config"`
-	SentryConfig            sentry.Config                       `json:"sentry_config"`
-	DeploymentConfig        DeploymentConfig                    `json:"deployment_config" validate:"required,dive"`
-	MessageQueueConfig      common_mq_config.MessageQueueConfig `json:"message_queue_config" validate:"required,dive"`
-	ManagementService       ManagementServiceConfig             `json:"management_service" validate:"required,dive"`
-	MonitoringConfig        Monitoring                          `json:"monitoring_config"`
-	SwaggerConfig           SwaggerConfig                       `json:"swagger_config" validate:"required,dive"`
-	SegmenterConfig         map[string]interface{}              `json:"segmenter_config"`
+	AssignedTreatmentLogger AssignedTreatmentLoggerConfig        `json:"assigned_treatment_logger"`
+	DebugConfig             DebugConfig                          `json:"debug_config" validate:"required,dive"`
+	NewRelicConfig          newrelic.Config                      `json:"new_relic_config"`
+	SentryConfig            sentry.Config                        `json:"sentry_config"`
+	DeploymentConfig        DeploymentConfig                     `json:"deployment_config" validate:"required,dive"`
+	MessageQueueConfig      *common_mq_config.MessageQueueConfig `json:"message_queue_config"`
```

Comment:
I would suggest to minimize using pointer, unless it is necessary, because you will need to do nil check before accessing the attribute everywhere in the code.

Reply:
This is changed to value instead of a pointer.